### PR TITLE
Homework 6

### PR DESCRIPTION
**Pull request by VShutkina**
Homework 6 - completed

### CODE REVIEW

File: src/test/java/hw6/ex2/Ex2CucumberHooks.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import cucumber.api.java.After;
+import cucumber.api.java.Before;
+import hw6.common.WebDriverSingleton;
+import io.github.bonigarcia.wdm.WebDriverManager;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+
+import java.util.concurrent.TimeUnit;
+
+public class Ex2CucumberHooks {
```
Comment (khda91) [inline src/test/java/hw6/ex2/Ex2CucumberHooks.java]: Code duplication

File: src/test/java/hw6/ex2/RunTests.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package hw6.ex2;
+
+import cucumber.api.CucumberOptions;
+import cucumber.api.testng.AbstractTestNGCucumberTests;
+
+@CucumberOptions(features = "classpath:hw6/ex2")
+public class RunTests extends AbstractTestNGCucumberTests {
```
Comment (khda91) [inline src/test/java/hw6/ex2/RunTests.java]: code duplication

File: src/test/java/hw6/ex2/steps/SecondExerciseActionSteps.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
+package hw6.ex2.steps;
+
+import cucumber.api.java.en.When;
+import hw6.common.AbstractBaseSteps;
+import hw6.ex2.TestStorage;
+
+public class SecondExerciseActionSteps extends AbstractBaseSteps {
```
Comment (khda91) [inline src/test/java/hw6/ex2/steps/SecondExerciseActionSteps.java]: Steps could be combined with ex1

File: src/test/java/hw6/ex2/steps/SecondExerciseAssertionSteps.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+import hw6.ex2.TestStorage;
+import org.testng.asserts.SoftAssert;
+
+import java.util.List;
+import java.util.Map;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+public class SecondExerciseAssertionSteps extends AbstractBaseSteps {
```
Comment (khda91) [inline src/test/java/hw6/ex2/steps/SecondExerciseAssertionSteps.java]: steps could be combined with the ex1

File: src/test/java/hw6/ex2/steps/SecondExerciseModalSteps.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package hw6.ex2.steps;
+
+import cucumber.api.java.en.Given;
+import hw6.HomePage;
+import hw6.common.AbstractBaseSteps;
+import hw6.common.WebDriverSingleton;
+
+public class SecondExerciseModalSteps extends AbstractBaseSteps {
```
Comment (khda91) [inline src/test/java/hw6/ex2/steps/SecondExerciseModalSteps.java]: code duplication

**Review by khda91 — CHANGES_REQUESTED** [review]
Please resolve comments

File: src/test/java/hw6/ex2/Ex2CucumberHooks.java (hunk 1 above)
Comment (VShutkina) [reply src/test/java/hw6/ex2/Ex2CucumberHooks.java]: fixed

File: src/test/java/hw6/ex2/RunTests.java (hunk 2 above)
Comment (VShutkina) [reply src/test/java/hw6/ex2/RunTests.java]: fixed

File: src/test/java/hw6/ex2/steps/SecondExerciseActionSteps.java (hunk 3 above)
Comment (VShutkina) [reply src/test/java/hw6/ex2/steps/SecondExerciseActionSteps.java]: fixed

File: src/test/java/hw6/ex2/steps/SecondExerciseAssertionSteps.java (hunk 4 above)
Comment (VShutkina) [reply src/test/java/hw6/ex2/steps/SecondExerciseAssertionSteps.java]: fixed

File: src/test/java/hw6/ex2/steps/SecondExerciseModalSteps.java (hunk 5 above)
Comment (VShutkina) [reply src/test/java/hw6/ex2/steps/SecondExerciseModalSteps.java]: fixed

**Review by khda91 — APPROVED** [review]
Approved. Could be merged into master branch

**Pull request merged**